runtime/jcl/common/criusupport.c

```diff
@@ -41,6 +103,200 @@ Java_com_ibm_jvm_CRIUSupport_isCRIUSupportEnabledImpl(JNIEnv *env, jclass unused
 		}
 #endif /* defined(LINUX) */
 	}
+	setupJNIFieldIDs(env);
 
 	return res;
 }
+
+static BOOLEAN
+getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeString, jfieldID *resultType)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	char mutf8StringBuf[STRING_BUFFER_SIZE];
+	char *mutf8String = NULL;
+	UDATA mutf8StringBufSize = 0;
+	UDATA requiredConvertedStringSize = 0;
+	BOOLEAN res = TRUE;
+	PORT_ACCESS_FROM_VMC(currentThread);
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+
+	mutf8String = vmFuncs->copyStringToUTF8WithMemAlloc(currentThread, javaString, J9_STR_NULL_TERMINATE_RESULT, "", 0, mutf8StringBuf, STRING_BUFFER_SIZE, &mutf8StringBufSize);
+	if (NULL == mutf8String) {
+		vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+		*resultType = vm->criuSupportJVMCheckpointFailure;
+		res = FALSE;
+		goto free;
+	}
+
+	/* get the required size */
+	requiredConvertedStringSize = omrstr_convert(J9STR_CODE_MUTF8, J9STR_CODE_PLATFORM_RAW,
+					mutf8String,
```

Comment:
Expect compiler warnings about this which passes `char *` where `uint8_t *` is expected; explicit casts may be required.

Reply:
omrport.h defines it as `int32_t (*str_convert)(struct OMRPortLibrary *portLibrary, int32_t fromCode, int32_t toCode, const char *inBuffer, uintptr_t inBufferSize, char *outBuffer, uintptr_t outBufferSize) ;`

Reply:
Ok, OMR is inconsistent; I was looking at https://github.com/eclipse/omr/blob/master/port/common/omrstr.c#L248 which says
```
omrstr_convert(struct OMRPortLibrary *portLibrary, int32_t fromCode, int32_t toCode,
			   uint8_t *inBuffer, uintptr_t inBufferSize, uint8_t *outBuffer, uintptr_t outBufferSize)
```